db/migrate/20240731194619_create_taggings.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+class CreateTaggings < ActiveRecord::Migration[7.1]
+  def up
+    create_taggings
+
+    safety_assured do
+      convert_status_taggings
+      convert_account_taggings
+    end
+
+    drop_table :accounts_tags
+    drop_table :statuses_tags
+  end
+
+  def down
+    restore_join_tables
+
+    safety_assured do
+      restore_account_taggings
+      restore_status_taggings
+    end
+
+    drop_table :taggings
+  end
+
+  def create_taggings
+    create_table :taggings do |t|
+      t.references :tag, null: false, foreign_key: true
+      t.references :taggable, polymorphic: true, null: false
+
+      t.timestamps
+    end
+  end
+
+  def restore_join_tables
+    create_table :statuses_tags, primary_key: [:tag_id, :status_id], force: :cascade do |t| # rubocop:disable Rails/CreateTableWithTimestamps
+      t.bigint :status_id, null: false
+      t.bigint :tag_id, null: false
+      t.index :status_id
+    end
+
+    create_table :accounts_tags, primary_key: [:tag_id, :account_id], force: :cascade do |t| # rubocop:disable Rails/CreateTableWithTimestamps
+      t.bigint :account_id, null: false
+      t.bigint :tag_id, null: false
+      t.index [:account_id, :tag_id]
+    end
+  end
+
+  def convert_status_taggings
+    execute <<~SQL.squish
+      INSERT INTO taggings(tag_id, taggable_type, taggable_id, created_at, updated_at)
+      SELECT statuses_tags.tag_id, 'Status', statuses_tags.status_id, statuses.updated_at, statuses.updated_at
```

Comment:
```suggestion
      SELECT statuses_tags.tag_id, 'Status', statuses_tags.status_id, statuses.created_at, statuses.updated_at
```

Reply:
Did you mean to update the `created_at` value to be the same as `updated_at`?

Reply:
> Did you mean to update the `created_at` value to be the same as `updated_at`?

I'm not sure what I meant ... and I also don't have a really strong opinion about which is more correct.

We currently dont have either datetime value on the habtm join table, but we will have it and be populating it going forward on the HMT taggings table. I think that backfilling the values with `NOW()` would be wrong, and I think either of the created_at or updated_at time for the account/status in question would be acceptable. Will gladly defer to anyone who feels strongly about one over the other.

Reply:
Oh I see, I was reading through this PR and I thought it was a typo. Thanks for the explanation.